src/plugins/discover/public/context_awareness/profile_providers/logs_data_source_profile/accessors/get_row_additional_leading_controls.ts

```diff
@@ -7,11 +7,24 @@
  */
 
 import { createDegradedDocsControl, createStacktraceControl } from '@kbn/discover-utils';
+import { retrieveMetadataColumns } from '@kbn/esql-utils';
 import type { DataSourceProfileProvider } from '../../../profiles';
+import { RowControlsExtensionParams } from '../../../types';
 
 export const getRowAdditionalLeadingControls: DataSourceProfileProvider['profile']['getRowAdditionalLeadingControls'] =
   (prev) => (params) => {
     const additionalControls = prev(params) || [];
+    const { query, dataView } = params;
 
-    return [...additionalControls, createDegradedDocsControl(), createStacktraceControl()];
+    const isDegradedDocsControlEnabled =
+      dataView.type === 'esql' ? queryContainsMetadataIgnored(query) : true;
```

Comment:
```suggestion
      isOfAggregateQueryType(query) ? queryContainsMetadataIgnored(query) : true;
```
Nit: I'd recommend using `isOfAggregateQueryType` for this instead since it's what we typically use, and it'll cast `query` to the correct type, so you could update the signature below to `queryContainsMetadataIgnored(query: AggregateQuery)`.